### PR TITLE
Adds --config option for configuration file

### PR DESCRIPTION
**Pull request by meghfossa**
# Overview

Adds --config option for fossa-cli.

## Acceptance criteria

- `--config` flag can be used with fossa analyze
- `--config` flag can be used with fossa test
- config file defaults to `.fossa.yml` file in target directory of analyze command or test command
- config file defaults to `.fossa.yml` file in current working directory, for any other command 

## Testing plan

```
echo 'version: 2' >> .fossa-dev-local.yml
echo 'version: 3' >> .fossa.yml

mkdir deepFolder
echo 'version: 1' >> deepFolder/.fossa-dev-deep.yml

# should not display warnings related to version number
./fossa analyze -o

# should display warnings related to version number 2
./fossa analyze -o --config .fossa-dev-local.yml

# should display warnings related to version number 1
./fossa analyze -o --config ./deepFolder/.fossa-dev-deep.yml

```

## Risks

N/A

## References

Closes https://github.com/fossas/team-analysis/issues/764

## Checklist

- [x] I added tests for this PR's change (or confirmed tests are not viable).
- [x] If this PR introduced a user-visible change, I added documentation into `docs/`.
- [x] I updated `Changelog.md` if this change is externally facing. If this PR did not mark a release, I added my changes into an `# Unreleased` section at the top.
- [x] I linked this PR to any referenced GitHub issues, if they exist.

### CODE REVIEW

File: docs/references/subcommands/report.md [hunk 1]
```diff
@@ -52,9 +52,10 @@ To use this compatibility script:
...
+| `--project 'some project'`         | `-p`  | Override the detected project name                                                                                                                     |
+| `--revision 'some revision'`       | `-r`  | -Override the detected project revision                                                                                                                |
+| `--fossa-api-key 'my-api-key'`     |       | An alternative to using the `FOSSA_API_KEY` environment variable to specify a FOSSA API key                                                            |
+| `--endpoint 'https://example.com'` | `-e`  | Override the FOSSA API server base URL                                                                                                                 |
+| `--config .fossa.dev.yml`          | `-c`  | Path to a [configuration file](/docs/references/files/fossa-yml.md) including filename. By default we look for `.fossa.yml` in base working directory. |
```
Comment (scruffystuffs) [inline docs/references/subcommands/report.md:61]: [nit]: Let's use the more conventional file example here: `/path/to/file`
```suggestion
| `--config /path/to/file`           | `-c`  | Path to a [configuration file](/docs/references/files/fossa-yml.md) including filename. By default we look for `.fossa.yml` in base working directory. |
```

File: src/App/Fossa/Main.hs [hunk 2]
```diff
@@ -150,14 +153,30 @@ mergeFileCmdConfig cmd file =
...
+          AnalyzeCommand AnalyzeOptions{analyzeBaseDir} -> do
+            baseDir <- sendIO $ validateDir analyzeBaseDir
+            pure (unBaseDir baseDir </> defaultConfigFile)
+          TestCommand TestOptions{testBaseDir} -> do
+            baseDir <- sendIO $ validateDir testBaseDir
+            pure (unBaseDir baseDir </> defaultConfigFile)
+          _ -> do
+            cwd <- getCurrentDir
+            pure (cwd </> defaultConfigFile)
+  fileConfig <- readConfigFileIO fileConfigPath
```
Comment (scruffystuffs) [inline src/App/Fossa/Main.hs:179]: Switching the behavior of where the default config file is located is a breaking change.  It's possible that this breaks existing workflows that rely on using the current working dir.

I'm not sure we actually want to make this change at the moment.  Unless we can get team agreement on breaking this, we should probably leave it as-is.  Also, it's not necessary for this PR, so if you decide you want to not wait for that discussion to get this merged, you could remove it and postpone it.

**Review by scruffystuffs — CHANGES_REQUESTED** [review]
Broadly, there's no issue with this change, but the part about changing the semantics of the default file has me concerned.  Also left a nit.

File: src/App/Fossa/Main.hs (hunk 2 above)
Comment (meghfossa) [reply src/App/Fossa/Main.hs]: I reverted to using cwd like before for config file in 599d282d9bf96e5ca52ffd787a76a404e4680b4a. 

I rather get this merged sooner, but I do think we should eventually look for config file at target directory of command, as opposed to cwd.

**Review by scruffystuffs — APPROVED** [review]
LGTM

**Pull request merged**